### PR TITLE
Add method to provide plugin information that is included in artifact zip

### PR DESCRIPTION
**Pull request by omkhegde**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: python/rpdk/go/parser.py [hunk 1]
```diff
@@ -7,9 +7,7 @@ def setup_subparser(subparsers, parents):
     parser.set_defaults(language="go")
 
     parser.add_argument(
-        "-p",
-        "--import-path",
-        help="Select the go language import path.",
+        "-p", "--import-path", help="Select the go language import path.",
```
Comment (anshikg) [inline python/rpdk/go/parser.py:10]: nit: can we remove formatting changes

**Review by anshikg — APPROVED** [review]

File: python/rpdk/go/parser.py (hunk 1 above)
Comment (omkhegde) [reply python/rpdk/go/parser.py]: Accidentally added these, removed now.

File: python/rpdk/go/codegen.py [hunk 2]
```diff
@@ -272,3 +272,10 @@ def write_with_relative_path(path):
         for path in (project.root / "internal").rglob("*"):
             if path.is_file():
                 write_with_relative_path(path)
+
+    @staticmethod
+    def _get_plugin_information():
+        return {"plugin-tool-version": __version__, "plugin-name": "go"}
```
Comment (wbingli) [inline python/rpdk/go/codegen.py:278]: How's the go plugin runtime version? or how go plugin manage the version?

Comment (omkhegde) [reply python/rpdk/go/codegen.py]: It's added in the go.mod file, for ex. -> https://github.com/brianterry/unicorn-maker/blob/master/go/go.mod#L7. Added code to parse the file and fetch version

**Review by wbingli — APPROVED** [review]

**Pull request merged**